From 5cb4099f3f48d5cbee613d8637898a862ebdc342 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 14 Feb 2022 08:45:06 +0200
Subject: [PATCH] [SW-75064] irq: rename ack irqs function

Various IRQs might be triggered during the reset sequence,
so the name "preboot_ack_razwi_irq" is too specific.
renamed to "preboot_ack_irq"

Change-Id: I3dcb4ab762af92a9328d7668edcf063d7224d3db
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi/arm/common.c b/arch/gaudi/arm/common.c
index 672fab0..10e49e0 100644
--- a/arch/gaudi/arm/common.c
+++ b/arch/gaudi/arm/common.c
@@ -533,7 +533,7 @@
 }
 
 #define GAUDI_IRQ_SIZE	512
-void preboot_ack_razwi_irq(void)
+void preboot_ack_irq(void)
 {
 	u32 i;
 
diff --git a/include/preboot.h b/include/preboot.h
index 44e2f74..b1b8a17 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -195,7 +195,7 @@
 void preboot_dev_enable_caps(void);
 void preboot_set_enabled_cap(enum hl_fw_en_caps en_cap);
 void preboot_finalize_enabled_caps(void);
-void preboot_ack_razwi_irq(void);
+void preboot_ack_irq(void);
 const char *preboot_get_version(enum preboot_vers ver);
 void hl_ram_init(void);
 void hl_pci_iatu_init(void);
diff --git a/lib/preboot/module_uart.c b/lib/preboot/module_uart.c
index 271f5f8..bbbecb6 100644
--- a/lib/preboot/module_uart.c
+++ b/lib/preboot/module_uart.c
@@ -78,17 +78,17 @@
 	preboot_uart_rbuf_init();
 }
 
-__weak void preboot_ack_razwi_irq(void) {}
+__weak void preboot_ack_irq(void) {}
 
 void module_uart_teardown(void)
 {
 	preboot_uart_rbuf_flush();
 
 	/*
-	 * RAZWI event might appear during boot due to async access
-	 * from host side to locked range registers (PCI) (i.e. during
-	 * iatu initlization or reset flow, PCI access is disabled).
-	 * Therefore this ack is needed.
+	 * Some events (like RAZWI or PCI ERR) might appear during boot
+	 * due to async access from host side to locked range registers (PCI)
+	 * (i.e. during iatu initlization or reset flow, PCI access is
+	 * disabled). Therefore this ack is needed.
 	 */
-	preboot_ack_razwi_irq();
+	preboot_ack_irq();
 }
